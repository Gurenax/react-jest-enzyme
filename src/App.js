import React, { Component } from 'react'
// import logo from './logo.svg';
import './App.css'

/* Components */
import Link from './components/Link'

class App extends Component {
  // constructor(props) {
  //   super(props)

  //   this.
  // }
  state = {
    clicked: false
  }

  onLinkClick = () => {
    this.setState({
      clicked: true
    })
  }
  
  render() {
    return (
      <div className="App">
        <h1>React Jest Enzyme</h1>
        <Link title="Google" url="http://www.google.com" onLinkClick={this.onLinkClick} />
        <button onClick={this.onLinkClick}></button>
      </div>
    )
  }
}

export default App
