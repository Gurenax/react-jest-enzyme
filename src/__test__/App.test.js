import React from 'react';
import ReactDOM from 'react-dom';
import { shallow, mount } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'
import App from '../App';

/*
*   Default test that comes with create-react-app
*/
it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

describe('App', () => {

  /*
  * Test by matching a snapshot
  */
  it('renders correctly', () => {
    const output = shallow(
      <App />
    )
    expect(shallowToJson(output)).toMatchSnapshot()
  })

  /*
  * Test by rendering through mounting
  */
  it('always renders a div', () => {
    const output = mount(
      <App />
    )
    const divs = output.find('div')
    expect(divs.length).toBeGreaterThan(0)
  })

  /*
  * Test by comparing the children
  */
  it('contains everything else that gets rendered', () => {
    const output = mount(
      <App />
    )
    const divs = output.find('div')
    // Outermost div is first on the list
    const outermostDiv = divs.first()
    // console.log(outermostDiv.children().getElements())
    // console.log(output.children().children().getElements())
    expect(outermostDiv.children()).toEqual(output.children().children())
  })

  /*
  *   Test when state changes with standard event
  */
  it('changes the state when button is clicked', () => {
    const output = shallow(<App />)
    // Before click
    expect(output.state().clicked).toEqual(false)
    // During click
    output.find('button').simulate('click') // Standard onClick
    // After click
    expect(output.state().clicked).toEqual(true)
  })

  /*
  *   Test when state changes with custom event 'onLinkClick'
  */
  it('changes the state when Link is clicked', () => {
    const output = shallow(<App />)
    // Before click
    expect(output.state().clicked).toEqual(false)
    // During click
    output.find('Link').simulate('linkClick') // Because event is called onLinkClick
    // After click
    expect(output.state().clicked).toEqual(true)
  })
})