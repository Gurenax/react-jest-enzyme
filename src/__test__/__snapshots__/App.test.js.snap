// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<div
  className="App"
>
  <h1>
    React Jest Enzyme
  </h1>
  <Link
    onLinkClick={[Function]}
    title="Google"
    url="http://www.google.com"
  />
  <button
    onClick={[Function]}
  />
</div>
`;
